grin/test/LintSpec.hs

```diff
@@ -224,45 +224,38 @@ spec = do
       let (_, errors) = lint allWarnings (Just typeEnv) program
       lintErrors errors `shouldBe` ["Invalid pattern match for (CInt x). Expected pattern of type: {CInt[T_Dead]}, but got: {CFloat[T_Float]}"]
 
-    it "disregards variable patterns" $ do
+    it "doesn't alert over-approximated binds" $ do
       let program = [prog|
           main =
-            n0 <- pure (CInt 0)
-            n1 <- case n0 of
-              (CInt c0) -> pure n0
-              (CFloat c1) ->
-                a0 <- pure (CFloat 2.0)
-                pure a0
+            i <- pure (CInt 1)
+            f <- pure (CFloat 1.0)
+            l <- store i
+            update l f
+            v <- fetch l
+            (CFloat f2) <- pure v
             pure ()
         |]
```

Comment:
Isn't this semantically invalid GRIN code? As in `store`, `fetch` and `update` must always come in this order for a given location.

Reply:
I am not sure. We could simply insert a fetch operation before the update, and that wouldn't change the result of the HPT.
@csabahruska What do you think?

Reply:
IMO it is not the point of this specific test so it is acceptable here.